cmake/FindRust.cmake

```diff
@@ -497,6 +498,14 @@ if (_RESOLVE_RUSTUP_TOOLCHAINS)
         rustc
             HINTS "${_RUST_TOOLCHAIN_PATH}/bin"
             NO_DEFAULT_PATH)
+elseif (Rust_RUSTUP)
```

Comment:
I think we do. In the particular system I'm working on, rustup is installed but not in PATH (this is why I wrote the previous fixes to invoke cargo/rustc by their full path). It gets discovered correctly by corrosion only because we hint at the rustup installation directory:
```
find_program(_Rust_COMPILER_TEST rustc PATHS "$ENV{HOME}/.cargo/bin")
```
If I try to find rustc without hinting at the directory where `rustup` was found, I'm afraid it will break systems where rustup is not in PATH.

Reply:
We might be able to write this nicer by moving `"$ENV{HOME}/.cargo/bin"` into a global variable that is used everywhere to hint `find_program`, but I'm kind of worried about introducing new HINT paths into various places in code that I don't really know

Reply:
> only because we hint at the rustup installation directory

Actually now that I'm seeing this, we should probably add `$ENV{CARGO_HOME}/bin` as a hint, in case the user set CARGO_HOME to something non-default. Would you add this in your PR, or should I create an issue and do it later.

Reply:
Probably the rustup searching can be made nicer, but we can also leave that for the future, since it is quite easy to cause regressions here. So you can consider this thread non-blocking for this PR.

Reply:
Do you wanna add it in addition to the existing HINT path? or replace it

Reply:
I would do "in addition". It could be that CARGO_HOME is not set for whatever reason.